cmd/template_cmd.go

```diff
@@ -421,61 +443,30 @@ func convertFlagParamsToCookbookVersions(c *cli.Context, cbvsIn string) (map[str
 	return result, nil
 }
 
-// convertFlagParamsToParameterValues returns the json representation for the given friendly input format of parameter-values assignation
-func convertFlagParamsToParameterValues(c *cli.Context, cbIn string) (map[string]string, error) {
-	result := map[string]string{}
-	for _, cb := range strings.Split(cbIn, ",") {
-		values := regexp.MustCompile(`(.*?)(:)(\w.*)`).FindStringSubmatch(cb)
-		if len(values) == 0 {
-			return nil, fmt.Errorf("invalid input parameter values format %s", cb)
-		}
-		name, _, value := values[1], values[2], values[3]
-		if _, found := result[name]; found {
-			return nil, fmt.Errorf("detected duplicated parameter name: %s", name)
-		}
-		result[name] = value
-	}
-	return result, nil
-}
-
-// convertFlagParamsToConfigurationAttributesFromFile returns the json representation of configuration attributes taken from the input file or STDIN
-func convertFlagParamsToConfigurationAttributesFromFile(c *cli.Context, casIn string) (map[string]interface{}, error) {
+// convertFlagParamsJsonFromFileOrStdin returns the json representation of parameters taken from the input file or STDIN
+func convertFlagParamsJsonFromFileOrStdin(c *cli.Context, dataIn string) (map[string]interface{}, error) {
 	var content map[string]interface{}
-	if casIn == "-" {
+	if dataIn == "-" {
 		// read from STDIN
-		logrus.Info("Please, write configuration parameters json formatted:")
-		buff := ""
-		scanner := bufio.NewScanner(os.Stdin)
-		for scanner.Scan() {
-			line := scanner.Text()
-			if len(line) == 0 {
-				logrus.Info("Processing...")
-				break
-			}
-			buff += line
-		}
-		if err := scanner.Err(); err != nil {
-			return nil, fmt.Errorf("cannot read from standard input: %v", err)
-		}
-
-		if err := json.NewDecoder(strings.NewReader(buff)).Decode(&content); err != nil {
-			return nil, fmt.Errorf("invalid json formatted attributes")
+		log.Info("Please, write parameters json formatted:")
+		if err := json.NewDecoder(os.Stdin).Decode(&content); err != nil {
+			return nil, fmt.Errorf("invalid json formatted parameter")
 		}
 	} else {
 		// read from file
-		sourceFilePath := casIn
+		sourceFilePath := dataIn
 		if !utils.FileExists(sourceFilePath) {
 			return nil, fmt.Errorf("invalid file path, no such file: %s", sourceFilePath)
 		}
 
-		attrsFile, err := os.Open(sourceFilePath)
+		jsonFile, err := os.Open(sourceFilePath)
 		if err != nil {
-			return nil, fmt.Errorf("cannot open file: %s. %v", sourceFilePath, err)
+			return nil, fmt.Errorf("cannot open file %s: %v", sourceFilePath, err)
 		}
-		defer attrsFile.Close()
+		defer jsonFile.Close()
 
-		if err = json.NewDecoder(attrsFile).Decode(&content); err != nil {
-			return nil, fmt.Errorf("invalid json formatted attributes in file %s", sourceFilePath)
+		if err = json.NewDecoder(jsonFile).Decode(&content); err != nil {
+			return nil, fmt.Errorf("invalid json formatted parameters in file %s", sourceFilePath)
```

Comment:
This code could be more compact and have the json.NewDecoder bit reused for both the STDIN and the normal file case, which is one of the reasons why I was requesting a refactor:
``` Golang
var reader io.Reader
var content map[string]interface{}
if dataIn == "-" {
  reader = os.Stdin
  dataIn = "STDIN"
} else {
  jsonFile, err := os.Open(dataIn)
  if err != nil {
    return nil, fmt.Errorf("cannot open %s to read JSON params: %v", dataIn, err)
  }
  defer jsonFile.Close()
  reader = jsonFile
}
err = json.NewDecoder(jsonFile).Decode(&content)
if err != nil {
  return nil, fmt.Errorf("cannot read JSON params from %s: %v", dataIn, err)
}
return content, nil
```

Reply:
Done!